customer-service/src/main/scala/domain/HttpClient.scala

```diff
@@ -0,0 +1,21 @@
+package domain
+
+import cats.effect.Async
+import io.circe.generic.auto._
+import org.http4s.Uri
+import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.client.{Client, JavaNetClientBuilder}
+
+class HttpClient[F[_]: Async](port: Int, host: String)(client: Client[F]) {
+
+  private val root                                     = Uri.unsafeFromString(s"http://$host:$port/internal")
+  def getClientDetails(id: String): F[CustomerDetails] = client.expect[CustomerDetails](root / id)
```

Comment:
u didn't separated yourself from external api, ur using same object and u drag it through layers

`def getClientDetails(id: String): F[CustomerDetails] = client.expect[CustomerDetailsExternalApiOutput](root / id)`

and u map `CustomerDetailsExternalApiOutput` into proper domain object: `CustomerDetails`

ofc `CustomerDetails` is not returned to your endpoint, u always have separation, so `CustomerDetails` in gateway should be mapped to `CustomerDetailsApiOutput`